feat: APP-374 header navigation toggle

r41ph opened the pull request:
## Description

https://regennetwork.atlassian.net/browse/APP-374

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### How to test

1. https://deploy-preview-2581--regen-marketplace.netlify.app/
2. check the navigation dropdown and the hamburger icon work on small screens as well as in desktop screens.
3. Both the hamburger icon navigation and the dropdown navigation should now:
- Close when the other one is opened.
- Toggle when clicked again.
- Close when open and a click occurs outside of them.
- Close when a nav item is clicked.

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

r41ph commented:
@erikalogie @S4mmyb see testing instructions

erikalogie commented:
LGTM

S4mmyb commented:
LGTM

blushi commented on web-components/src/components/header/components/HeaderMenuItem/HeaderMenuItem.Hover.tsx:
```diff
@@ -32,10 +32,15 @@ const HeaderMenuItemHover = ({
   const { classes: styles } = useMenuHoverStyles();
-
+  // eslint-disable-next-line lingui/no-unlocalized-strings
+  const isTouchScreen = useMediaQuery('(pointer: coarse)');
```
since we want to move away from MUI eventually, could we avoid using it here to detect touch screens?

r41ph replied:
Why change it? we use `useMediaQuery` is many other places so eventually we will have to create our own version.

blushi replied:
I was assuming we would not necessarily try to replicate `useMediaQuery` in the exact same way when migrating but potentially use different APIs, but I guess we'll see, ok to keep it as is for now

blushi commented:
Looks like when clicking on one of the user menu items (eg portfolio) after **initial page loading**, the dropdown menu appears in the bottom right corner for a few seconds (see video below).
Looks related to https://regennetwork.atlassian.net/browse/APP-28 but so far it was hard to reproduce, now with the changes in this PR, I can reproduce it easily using the instructions just above, so maybe it would be worth trying to tackle that?

r41ph commented:
@blushi see updated code please. I have refactored the header navigation dropdown to not use Popover and also amend some semantic issues. The dropdown now shouldn't jump anymore to the bottom right corner. I have also updated the `MobileMenu` so we don't have duplicated markup in the DOM. 

In some route changes like between `/`, `/projects`, `/dashboard` or `/admin/dashboard` I see like a flickering, like the whole app re-renders for some reason. I think we should look into this, have you noticed? wdyt?

r41ph commented:
> the dropdown menus are too large with some extra left padding

@blushi I'm not seeing the extra padding nor the missing colours. What browser are you using? Have you tried clearing the cache and doing hard refresh to see if it still happens?

> when opening the menu, the transition was a bit longer before

I'll review animation timing

blushi commented:
> > the dropdown menus are too large with some extra left padding
> 
> @blushi I'm not seeing the extra padding nor the missing colours. What browser are you using? Have you tried clearing the cache and doing hard refresh to see if it still happens?
> 
Yeah this doesn't help, using Chrome. I see the same on Brave, Firefox and Safari

r41ph commented:
@blushi I have updated the animation timing and also fixed the padding issue. Re the missing colours, I can't reproduce it in any browser anymore after seeing it in Brave once. Is it still happening to you in the latest deploy preview?

blushi reviewed (commented):
![image](https://github.com/user-attachments/assets/97ef9b35-41c4-4064-b699-e5b43448680d)
 the user menu layout at the top is wrong now
versus on dev: 
![image](https://github.com/user-attachments/assets/8d687ae1-326d-4548-bf69-9404b1f55456)

The animation feels less "smooth" than before, especially when the dropdown menu closes. Before the full dropdown was fading out, now it's using translateY/scale which looks a bit different.

r41ph commented:
> The animation feels less "smooth" than before, especially when the dropdown menu closes. Before the full dropdown...

@blushi, I have updated the timing again and remove the `translate(), wdyt?
@erikalogie @S4mmyb do you want to have another look?

S4mmyb commented:
Hey @r41ph , this looks good on mobile. One thing I noticed on desktop, though, is that for the user profile dropdown you can open the drop-down on-hover and collapse it with a click. But you can't get it to drop down again on-click, you need to move the mouse and hover on it again. 

https://github.com/user-attachments/assets/a8ad485c-b352-49b7-8ad9-39b08442844c

r41ph commented:
@S4mmyb the menu shouldn't close on click now (on desktop)

erikalogie commented:
LGTM

blushi reviewed (approved):
tACK
